### ISSUE TITLE
I'd like to ask a question about the resource consumption of bot-sort-reid.

### ISSUE THREAD

**Issue by isakkk3** [ultralytics/ultralytics#22407]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Hello, I would like to know about the resource usage rate of deploying bot-sort-reid on the embedded device RK3588. Can it run smoothly on RK3588 and achieve real-time tracking?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @isakkk3, thank you for your interest in Ultralytics 🚀! This is an automated response — an Ultralytics engineer will follow up soon to assist further. In the meantime, we recommend a visit to the [Docs](https://docs.ultralytics.com/) where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

For your RK3588 and BOT-SORT-ReID real-time tracking ❓ Question, please share the following details to help us provide accurate guidance:
-Exact Ultralytics version (`ultralytics`), model (e.g., YOLO11 variant), and tracker settings used (BOT-SORT-ReID config)
-Input details: resolution, expected FPS, number of streams/cameras, and typical scene complexity
-Hardware specifics: RK3588 clocking, available RAM/VRAM, and whether you leverage NPU/GPU acceleration
-Runtime configuration: batch size, precision (FP32/FP16/INT8), and any quantization or TensorRT/OpenVINO usage
-Logs or short profiling results (init time, per-frame latency for detection and ReID, CPU/GPU/NPU utilization)

Also verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) if you are training custom models for your tracker.

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [Python>=3.8](https://www.python.org/) environment with [PyTorch>=1.8](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

-**Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
-**Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
-**Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
-**Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by isakkk3** [comment]
The yolov5 model is used, the input is 1080p, the hardware is rk3588, quad-core ARM Cortex-A76@2.4GHz + quad-core ARM Cortex-A55@1.8GHz large and small core architecture.

**Comment by glenn-jocher** [comment]
Thanks for the details. Short answer: RK3588 can run YOLO + tracking, but BoT-SORT with ReID is CPU-bound and will be your bottleneck; 30 FPS at 1080p is unlikely. For smoothness on RK3588, export a small detector to RKNN, run at `imgsz=640`, consider disabling ReID or switching to ByteTrack, and if needed skip frames with `vid_stride=2`. See the Rockchip RKNN guide and tracking docs for settings. ([docs.ultralytics.com](https://docs.ultralytics.com/integrations/rockchip-rknn/))

Example:
```bash
# Export on an x86 Linux PC
yolo export model=yolo11n.pt format=rknn name=rk3588

# Track on RK3588 (BoT-SORT; toggle ReID in botsort.yaml via `with_reid`)
yolo track model=./yolo11n_rknn_model source=your.mp4 imgsz=640 tracker=botsort.yaml persist=True vid_stride=2
```
On RK3588 we measured YOLO11n RKNN ≈71.5 ms/image at 640 for detection-only (not including tracking/ReID). Details and benchmarks: see the Rockchip RKNN export page; BoT-SORT/ReID configuration (e.g., `with_reid`) is documented in the tracking guide. ([docs.ultralytics.com](https://docs.ultralytics.com/integrations/rockchip-rknn/))

- Rockchip export: [Rockchip RKNN export guide](https://docs.ultralytics.com/integrations/rockchip-rknn/)  
- Tracking and ReID settings: [Ultralytics tracking docs](https://docs.ultralytics.com/modes/track/)
